src/Metalama.Patterns.NotifyPropertyChanged/Implementation/Natural/NaturalAspect.cs

```diff
@@ -0,0 +1,392 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Advising;
+using Metalama.Framework.Aspects;
+using Metalama.Framework.Code;
+using Metalama.Framework.Code.DeclarationBuilders;
+using Metalama.Framework.Code.SyntaxBuilders;
+using Metalama.Framework.Eligibility;
+using Metalama.Framework.Engine.CodeModel;
+
+namespace Metalama.Patterns.NotifyPropertyChanged.Implementation.Natural;
+
+internal sealed partial class NaturalAspect : IAspect<INamedType>
```

Comment:
This class should not be an IAspect because of licensing and clarity reasons. It should be a ITemplateProvider. Use IAdviceFactory.WithTemplateProvider(this)